python/lsst/meas/algorithms/psfDeterminer.py

```diff
@@ -19,25 +19,56 @@
 # the GNU General Public License along with this program.  If not,
 # see <http://www.lsstcorp.org/LegalNotices/>.
 #
-from lsst.pex.config import makeRegistry
-from .pcaPsfDeterminer import PcaPsfDeterminer
+from __future__ import absolute_import, division, print_function
 
-__all__ = ["psfDeterminerRegistry"]
+import abc
 
-psfDeterminerRegistry = makeRegistry(
-    '''A registry of PSF determiner factories
+import lsst.pipe.base as pipeBase
+import lsst.pex.config as pexConfig
 
-    A PSF determiner factory makes a class with the following API:
+__all__ = ["BasePsfDeterminerConfig", "BasePsfDeterminerTask", "psfDeterminerRegistry"]
 
-    def __init__(self, config, schema=None):
+class BasePsfDeterminerConfig(pexConfig.Config):
+    kernelSize = pexConfig.Field(
```

Comment:
I only put in the few settings that were clearly needed by both existing PSF determiners and looked likely to be needed by all others, since we prefer to avoid fields that are ignored. I was disappointed that the PCA psf determiner has nothing like `badMaskBits`, since that would be an obvious shared config field. I'm not sure if `spatialOrder` or cell size would be universal. We can easily move more fields into this common config later when we are confident they are shared.